Documentation/contributors/design-proposals/webhooks.md

```diff
@@ -0,0 +1,115 @@
+
+### Truncated CSV example that includes webhook configuration
+
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - "*"
+            ...
+      permissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - ""
+            ...
+      deployments:
+        - name: webhook-deployment
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                name: mywebhook
+            template:
+              metadata:
+                labels:
+                  name: mywebhook
+              spec:
+                serviceAccountName: my-operator
+                containers:
+                - name: webhook
+                  image: operator-framework/fake-webhook-image:latest
+                  imagePullPolicy: IfNotPresent
+                  ports:
+                  - containerPort: 443
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  volumeMounts:
+                    - mountPath: /keys
+                      name: tls-keys
+                      readonly: true
+                - name: operator
+                  image: operator-framework/fake-operator:latest
+                  imagePullPolicy: IfNotPresent
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  - name: MY_POD_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.name
+                volumes:
+                - name: tls-keys
+                    secretName: service-tls
+      services:
+      - name: mywebhook-service
+        namespace: test-ns
+        spec:
+          selector:
+            name: mywebhook
+          ports:
+          - protocol: TCP
+            port 443
+            targetPort: 443
+      validatingWebhooks: # also available, mutatingWebhooks
+      - name: olm-admission
+        webhooks:
+          - name: csv.operators.olm.com
+          rules:
+            - apiGroups:
+              - "operators.olm.com"
+              apiVersions:
+              - v1alpha1
+              operations:
+              - CREATE
+              resources:
+              - clusterserviceversions
+          clientConfig:
+            caBundle: ## have this automatically created if not specified ##
```

Comment:
For APIServices, OLM always generates CA bundles. We could allow users to specify them, but we would also need to provide a way to specify the webhook's serving cert (or the CA's private key to generate it). We can let them reference secrets, but they would need to be created outside of OLM's control, or provided by a CatalogSource (the latter has its own set of issues).

Reply:
I pushed before I saw these comments. I didn't realize that the caBundle field would be "filled in" from the apiserver if unspecified. This to me seems like the behavior we'd want for an OLM-managed webhook.